rules/android_binary_internal/r8.bzl

```diff
@@ -73,9 +73,11 @@ def process_r8(ctx, jvm_ctx, packaged_resources_ctx, build_info_ctx, **_unused_c
 
     android_jar = get_android_sdk(ctx).android_jar
     proguard_specs = proguard.get_proguard_specs(ctx, packaged_resources_ctx.resource_proguard_config)
+    proguard_mappings_file = ctx.actions.declare_file(ctx.label.name + "_proguard.map")
```

Comment:
Native android_binary generates the mapping file only if `proguard_generate_mapping` is enabled: https://bazel.build/reference/be/android#android_binary.proguard_generate_mapping